Fix commands with pnpm's `--filter` in `CONTRIBUTING.md` (#1908)
iTwin/iTwinUI: CONTRIBUTING.md

If a script is not available in the root package.json or if you need to pass workspace-specific cli args, then you can specify the workspace as follows:
```
# passing Alert as a cli arg to the `test` command in itwinui-react

> pnpm --filter=@itwin/itwinui-react test Alert

Reviewer: ```suggestion
pnpm --filter=@itwin/itwinui-react run test Alert
```

Note that this bypasses the turborepo pipeline, so you will need to manually run any dependent tasks first. For example, if the `build` command of `react-workshop` relies on the `build` command of `@itwin/itwinui-react`, then you will need to manually run the `build` commands in the right order.
```
pnpm --filter=@itwin/itwinui-react build
pnpm --filter=react-workshop build
```

This is why it's recommended to use the turbo `--filter` syntax whenever possible.
```
pnpm run build --filter=react-workshop
```

> [!NOTE]
>
> The `--filter` argument is used in both `turbo` and `pnpm`. To use it with:
> * `turbo`: `pnpm run [command] --filter=[workspace]`
>   * e.g. `pnpm run build --filter=react-workshop`
> * `pnpm`: `pnpm --filter=[workspace] run [command]`

> >   * e.g. `pnpm --filter=react-workshop run build`

Author: > This looks good, but to make it even better we could add a short note (under "Running bespoke commands") to disambiguate between Pnpm's filter syntax and Turbo's filter syntax.

@mayank99 Is this like what you suggested?

Reviewer: Can add links and explain the difference.

```suggestion
> [!NOTE]
>
> The `--filter` syntax is available in both `turbo` and `pnpm`. The usage looks slightly different:
> - `turbo`: `pnpm run [command] --filter=[workspace]`
>   - e.g. `pnpm run build --filter=@itwin/itwinui-react`
>   - This will also run any dependent tasks defined in the Turbo pipeline, but does not allow args. (See [Turborepo docs](https://turbo.build/repo/docs/core-concepts/monorepos/filtering)).
> - `pnpm`: `pnpm --filter=[workspace] run [command] [args]`
>   - e.g. `pnpm --filter=@itwin/itwinui-react run test Alert`
>   - This will only run the task per-workspace and supports additional args. (See [Pnpm docs](https://pnpm.io/filtering)).
```

Note that this bypasses the turborepo pipeline, so you will need to manually run any dependent tasks first. For example, if the `build` command of `react-workshop` relies on the `build` command of `@itwin/itwinui-react`, then you will need to manually run the `build` commands in the right order.
```

> pnpm --filter=@itwin/itwinui-react build
> pnpm --filter=react-workshop build

Reviewer: ```suggestion
pnpm --filter=@itwin/itwinui-react run build
pnpm --filter=react-workshop run build
```

1. Make sure you have [Docker](https://www.docker.com/get-started) installed and running.
2. From the monorepo root, run `pnpm run test --filter=react-workshop`. This will build react-workshop and run all cypress tests in docker.

>    -  If you only need to run tests for a specific component, you can do so by passing the `--spec` argument to cypress. e.g. for testing `Alert`, you can run `pnpm --filter=react-workshop run test --spec="**/Alert.*"`. But don't forget to build react-workshop first (pnpm run build --filter=react-workshop).

Reviewer 2: Thanks for updating this!

In the Cypress GUI:

> 1. From the monorepo root, run `pnpm run --filter=a11y open`. This will open the Cypress control panel where you can run the tests.

Reviewer: ```suggestion
1. From the monorepo root, run `pnpm --filter=a11y run open`. This will open the Cypress control panel where you can run the tests.
```
